src/sync/browser.js

```diff
@@ -0,0 +1,105 @@
+import PushManagerFactory from './PushManager';
+import { PUSH_DISCONNECT, PUSH_CONNECT } from './constants';
+import FullProducerFactory from '../producer';
+import PartialProducerFactory from '../producer/browser/Partial';
+import { matching } from '../utils/key/factory';
+import { forOwn } from '../utils/lang';
+import logFactory from '../utils/logger';
+const log = logFactory('splitio-sync:sync-manager');
+
+/**
+ * Factory of sync manager for browser
+ *
+ * @param context main client context
+ */
+export default function BrowserSyncManagerFactory(mainContext) {
+
+  const contexts = {};
+  let pushManager;
+
+  function startPolling() {
+    log.info('PUSH down or disconnected. Starting periodic fetch of data.');
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
+      if (producer && !producer.isRunning())
+        producer.start(true); // `fetchers` are scheduled but not called immediately
+    });
+  }
+
+  function stopPollingAndSyncAll() {
+    log.info('PUSH (re)connected. Syncing and stopping periodic fetch of data.');
+    // if polling, stop
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
+      if (producer && producer.isRunning())
+        producer.stop();
+    });
+    syncAll();
+  }
+
+  function syncAll() {
+    // fetch splits and segments
+    const mainProducer = mainContext.get(mainContext.constants.PRODUCER, true);
+    mainProducer && mainProducer.synchronizeSplits();
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
+      producer && producer.synchronizeMySegments();
+    });
+  }
+
+  function shared(context, isSharedClient = true) {
```

Comment:
This should not be called shared, also the boolean flag is not necessary, specially given the form of the assignment in line #103. 
Invert the parameters, to create the instance you return as main one just call it with (false, context). Then in #103 do (using an example name) `syncManager.shared = createInstance.bind(null, true)` and just call the .shared() method from outside. 

Using a "shared" that creates the full instance and then reexporting as shared property on the instance itself, forcing a boolean telling IT IS a shared instance even though the property accessor is "shared"... seems like entangled.